### DISCUSSION TITLE
problem

### DISCUSSION THREAD

**Discussion by xianglingzhang** [ZenGo-X/multi-party-ecdsa#106] [General]
Here is the error I met when run "cargo +nightly build --release --examples"
error: linking with `cc` failed: exit code: 1
I can't solve this problem~~

**Comment by omershlo** [comment]
Hi @zhangxiangling-code , 
what operating system are you using ?

**Reply by xianglingzhang** [reply]
Linux

**Reply by omershlo** [reply]
what distribution ? 
do you have an option to try another OS?

**Reply by xianglingzhang** [reply]
Ubuntu 18.04, I have tried Ubuntu 20.04, still met the problem.
"error: linking with `cc` failed: exit code: 1"
"error: could not compile `multi-party-ecdsa`"

**Comment by omershlo** [comment]
Can you provide the full error you get ?

**Reply by xianglingzhang** [reply]
Compiling multi-party-ecdsa v0.4.3 (/home/zhang/multi-party-ecdsa-master)
error: linking with `cc` failed: exit code: 1
  |
  = note: "cc" "-Wl,--as-needed" "-Wl,-z,noexecstack" "-m64" "-Wl,--eh-frame-hdr" "-L" "/home/zhang/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "/home/zhang/multi-party-ecdsa-master/target/release/examples/gg20_sign_client-d926dd39ee1d1506.gg20_sign_client.235x6j26-cgu.0.rcgu.o" "/home/zhang/multi-party-ecdsa-master/target/release/examples/gg20_sign_client-d926dd39ee1d1506.gg20_sign_client.235x6j26-cgu.1.rcgu.o" "/home/zhang/multi-party-ecdsa-master/target/release/examples/gg20_sign_client-d926dd39ee1d1506.gg20_sign_client.235x6j26-cgu.10.rcgu.o" "/home/zhang/multi-party-ecdsa-master/target/release/examples/gg20_sign_client-d926dd39ee1d1506.gg20_sign_client.235x6j26-cgu.11.rcgu.o" "/home/zhang/multi-party-ecdsa-master/target/release/examples/gg20_sign_client-d926dd39ee1d1506.gg20_sign_client.235x6j26-cgu.12.rcgu.o" "/home/zhang/multi-party-ecdsa-master/target/release/examples/gg20_sign_client-d926dd39ee1d1506.gg20_sign_client.235x6j26-cgu.13.rcgu.o" "/home/zhang/multi-party-ecdsa-master/target/release/examples/gg20_sign_client-d926dd39ee1d1506.gg20_sign_client.235x6j26-cgu.14.rcgu.o" "/home/zhang/multi-party-ecdsa-master/target/release/examples/gg20_sign_client-d926dd39ee1d1506.gg20_sign_client.235x6j26-cgu.15.rcgu.o" "/home/zhang/multi-party-ecdsa-master/target/release/examples/gg20_sign_client-d926dd39ee1d1506.gg20_sign_client.235x6j26-cgu.2.rcgu.o" "/home/zhang/multi-party-ecdsa-master/target/release/examples/gg20_sign_client-d926dd39ee1d1506.gg20_sign_client.235x6j26-cgu.3.rcgu.o" "/home/zhang/multi-party-ecdsa-master/target/release/examples/gg20_sign_client-d926dd39ee1d1506.gg20_sign_client.235x6j26-cgu.4.rcgu.o" "/home/zhang/multi-party-ecdsa-master/target/release/examples/gg20_sign_client-d926dd39ee1d1506.gg20_sign_client.235x6j26-cgu.5.rcgu.o" "/home/zhang/multi-party-ecdsa-master/target/release/examples/gg20_sign_client-d926dd39ee1d1506.gg20_sign_client.235x6j26-cgu.6.rcgu.o" "/home/zhang/multi-party-ecdsa-master/target/release/examples/gg20_sign_client-d926dd39ee1d1506.gg20_sign_client.235x6j26-cgu.7.rcgu.o" "/home/zhang/multi-party-ecdsa-master/target/release/examples/gg20_sign_client-d926dd39ee1d1506.gg20_sign_client.235x6j26-cgu.8.rcgu.o" "/home/zhang/multi-party-ecdsa-master/target/release/examples/gg20_sign_client-d926dd39ee1d1506.gg20_sign_client.235x6j26-cgu.9.rcgu.o" "-o" "/home/zhang/multi-party-ecdsa-master/target/release/examples/gg20_sign_client-d926dd39ee1d1506" "/home/zhang/multi-party-ecdsa-master/target/release/examples/gg20_sign_client-d926dd39ee1d1506.4n12i90m9z4lt6qn.rcgu.o" "-Wl,--gc-sections" "-pie" "-Wl,-zrelro" "-Wl,-znow" "-Wl,-O1" "-nodefaultlibs" "-L" "/home/zhang/multi-party-ecdsa-master/target/release/deps" "-L" "/home/zhang/multi-party-ecdsa-master/target/release/build/secp256k1-3d6e85b7cc31acef/out" "-L" "/home/zhang/multi-party-ecdsa-master/target/release/build/rust-crypto-4d7abcdacf58b494/out" "-L" "/home/zhang/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib" "-Wl,-Bstatic" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libhex-f0ac4054f3e10cf0.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libcrypto-db9d60e65b339a93.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/librustc_serialize-aa361dce80a6c671.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/librand-c0f23b75a5dad1ae.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/librand-143f562f8512c5b5.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libreqwest-6dea1779f9dab245.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libuuid-7e48438dd3600c71.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libserde_urlencoded-1a89f86c0f4287b4.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libdtoa-2447b87b20875ed4.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libmime_guess-90ce58d79a8765e7.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libunicase-420baa79e66eb7be.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libmime-0c405423f2f6d82d.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libflate2-749aef3ac060c4b7.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libcrc32fast-3fc58e0e7ee06998.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libhyper-e47e677a16205970.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libwant-7c424922b481dbc5.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libtry_lock-0d0e456c3c6f815b.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libtokio-1802b11e52a9380a.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libtokio_timer-c47dea94e8f10ca7.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libtokio_threadpool-b8358d33b7fe8ce6.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libcrossbeam_queue-2b71bc23b650b26b.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libcrossbeam_deque-3b1404a4135346d6.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libcrossbeam_epoch-ac98a8ea1c0debef.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libmemoffset-31fb3f8caba07ffe.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libtokio_tcp-d2695ecea0e92550.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libtokio_reactor-b29b44862439e836.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libtokio_sync-a0680c79ac40ae0b.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libparking_lot-ad46b3da2cebcd0b.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libparking_lot_core-e2b31b5084839e13.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libsmallvec-1f2539e33263dc88.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libmaybe_uninit-56e7042b6857a274.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/liblock_api-d528f67b95c65e90.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libtokio_current_thread-8fd7c9c41490e9c7.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libtokio_executor-94eda104bf695200.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libcrossbeam_utils-711e824a73a463d6.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libmio-1ac2c4f018826c2a.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libnet2-ef93578195259743.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libhttparse-6f0db3eff7258637.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libhttp_body-0306b4c79db18ee0.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libtokio_buf-c49c30c7df37cfd8.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libh2-19d9c789eaccbeb8.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libindexmap-1e55b735637bcc6a.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libhashbrown-c723dee100e54a5e.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libstring-da15d8d5fd0152b2.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libslab-314cf7e98398e4c0.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libtokio_io-fa37c4c857b62ef9.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libfutures_cpupool-8ea1fc14f69fc4fd.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libhttp-70f14a146e27efae.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libfnv-b63504f5018cc108.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libfutures-544643fbe27daa4c.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libencoding_rs-9c06080874b247fd.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libcookie_store-20e0c8ac83ab8129.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/liblog-fe5eb1ec3d35d5a3.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libpublicsuffix-d48d8c6edae5889a.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/liburl-bdb51ad60a81f567.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libform_urlencoded-6c027baec4c4b4b3.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libpercent_encoding-a93af9c56752979f.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libidna-2cff8351c96b71bb.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libregex-0254e3d4c6cc0fb1.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libregex_syntax-54ff6446d5044f96.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/liberror_chain-d66f4a943995ab41.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libtry_from-d2a14833014f4b6b.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libfailure-954018d50bd07fa8.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libbacktrace-3804904f9042f2f6.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libminiz_oxide-40186c75165dddf1.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libadler-392f602eb35377a4.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libobject-1337fbe988daf54a.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libaddr2line-dc9975fc9ab8c2d7.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libgimli-da39de18b8474960.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/librustc_demangle-a1486b6107e2bf83.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libcookie-c8854dd991f425b6.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libtime-9445a996f553a818.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/liburl-b126357f4d0d4ecf.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libpercent_encoding-027fc70817b8b184.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libidna-ee1ff6ebcefa9e2b.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libunicode_normalization-815d148452844814.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libtinyvec-295d5ad3fc0cbc9c.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libtinyvec_macros-7830452eb45edc19.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libunicode_bidi-41abae14da234d1b.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libmatches-32581ec9f6412ebd.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libbytes-f74677bda2712f23.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libiovec-1f66aaff7ad3d529.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libbase64-09c3479e8c2b0262.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libmulti_party_ecdsa-dc5f11702d4a5edb.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libsecp256k1-213483a5a237d074.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/librand-126fde7ba370b335.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/librand_chacha-ce81d2640257ac2a.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libppv_lite86-a6feee9ed6f337c0.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/librand_core-f7c8830cd97c0333.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libgetrandom-bde58686f06c8460.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libcfg_if-e0361196973a0e81.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libhmac_drbg-f18ac4789a637bc3.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libcrunchy-a1a63dd0e5d691b9.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libsubtle-55b44176117a0513.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libzk_paillier-ba26719c4e0e3aa5.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libbit_vec-cb877688d34f8b67.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libpaillier-5fb2e8d32c0fa3b0.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libcurv-ef971cb5605cf2a7.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libcentipede-cf6c01dd535fc004.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/librayon-5605f280aa49813c.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/librayon_core-80dc8c43cc5c85e7.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libnum_cpus-1ea82628b39098eb.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libcrossbeam_deque-44c966133930651c.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libcrossbeam_epoch-210f67111ea42c0d.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libmemoffset-ff7d393a4e864cac.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libscopeguard-680be5d68e26570b.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libcrossbeam_channel-06ec3e5bab63dd2d.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libcrossbeam_utils-1f30739d8f8db769.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/liblazy_static-72fd4c50b9d3882e.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libcfg_if-cb3a90c6c918d108.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libbulletproof-2b43c9be90a3533a.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libitertools-f3b6946d125c0e6f.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libeither-f3bc76f3ccc8e9f2.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libcurv-6641d2426246dae3.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libhmac-9d85fffb0d093356.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libcrypto_mac-cf605242fc7734e3.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libsubtle-44a12f187f6d2fc6.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libsha2-a965625e9b4b1d77.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libfake_simd-3756b16c63a82254.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libblake2b_simd-bce441d9cc6d9c01.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libconstant_time_eq-5f4929594022016d.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libarrayvec-c8bff44e6ceb9a6a.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libarrayref-fa8491c85ad5e872.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libsha3-83c5ca7a5dd714c1.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libdigest-30da3c50b4689178.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libopaque_debug-b3511308285898cb.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libblock_buffer-ec72f459c0678688.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libgeneric_array-8c74a886c0bfa1b8.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libtypenum-8754dce572e25207.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libblock_padding-2ed51b72affadbdd.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libbyte_tools-212432e5188e0eb9.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libbyteorder-696e60eb23609a9a.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libkeccak-f2022e5e0c1cc304.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libgmp-f82c97949c9a313a.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libserde_json-ca42dbfdda55174b.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libryu-b04ad59bf2ec6b7b.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libitoa-c64313853daac3b2.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libnum_traits-2d6f0e657d8b410e.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libzeroize-24347eb4bd7544f0.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libsecp256k1-d9d968f2c68152b6.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/libserde-3c590a4fc7e3e3b1.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/librand-5b42d383e54c44bd.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/librand_xorshift-3ddf70d45335aa6b.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/librand_pcg-bb49864450bf3cd7.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/librand_hc-7e866376272e962a.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/librand_chacha-c4d61c2b1a62a54b.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/librand_isaac-a29e978858277bbc.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/librand_core-4d2b942fff69a04a.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/librand_os-7cdfb5509b25a08b.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/liblibc-ee7fc30d9ad975a5.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/librand_jitter-ac589ba51e74fb53.rlib" "/home/zhang/multi-party-ecdsa-master/target/release/deps/librand_core-a3b56790fe374005.rlib" "-Wl,--start-group" "/home/zhang/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libstd-25c6acf8063a3802.rlib" "/home/zhang/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libpanic_unwind-539f13c9442f1597.rlib" "/home/zhang/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libminiz_oxide-37db28e905edb56b.rlib" "/home/zhang/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libadler-9ef2480568df55af.rlib" "/home/zhang/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libobject-1e0f0992cdbecd66.rlib" "/home/zhang/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libaddr2line-6c8e02b8fedc1e5f.rlib" "/home/zhang/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libgimli-807e5ad203594490.rlib" "/home/zhang/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_demangle-083fce1bea11612a.rlib" "/home/zhang/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libhashbrown-1af568081add9042.rlib" "/home/zhang/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_std_workspace_alloc-1395b54a3b3f45bf.rlib" "/home/zhang/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libunwind-caba820045f178d5.rlib" "/home/zhang/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcfg_if-512eb53291f6de7e.rlib" "/home/zhang/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liblibc-5efacc5025f9f3d8.rlib" "/home/zhang/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/liballoc-9c4002b5f79ba0e1.rlib" "/home/zhang/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/librustc_std_workspace_core-90996f4879673567.rlib" "/home/zhang/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcore-9ea09a899c3eda46.rlib" "-Wl,--end-group" "/home/zhang/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/lib/libcompiler_builtins-ef2408da76957905.rlib" "-Wl,-Bdynamic" "-lgmp" "-lgmp" "-lgmp" "-lgmp" "-lgmp" "-lgcc_s" "-lutil" "-lrt" "-lpthread" "-lm" "-ldl" "-lc"
  = note: /usr/bin/ld: cannot find -lgmp
          /usr/bin/ld: cannot find -lgmp
          /usr/bin/ld: cannot find -lgmp
          /usr/bin/ld: cannot find -lgmp
          /usr/bin/ld: cannot find -lgmp
          collect2: error: ld returned 1 exit status
          

error: aborting due to previous error

error: could not compile `multi-party-ecdsa`

**Comment by omershlo** [comment]
Ok, you need to install gmp:  , it’s the library we use for big integers , it is supposed to be installed as part of the distribution- not sure what is special about your OS but I will keep track 
https://gmplib.org/

**Reply by xianglingzhang** [reply]
Thanks!!! Problem is solved!
